src/core/resolvers/require.zig

```diff
@@ -120,14 +132,36 @@ pub fn zune_require(L: *VM.lua.State) !i32 {
     var outErr: ?[]const u8 = null;
     var moduleRelativePath: [:0]const u8 = undefined;
     var searchResult: ?file.SearchResult([:0]const u8) = null;
```

Comment:
Cases anywhere in between this variable and the next `defer`, errors would cause a leak. 
```suggestion
    var searchResult: ?file.SearchResult([:0]const u8) = null;
    defer if (searchResult) |r| r.deinit();
```